### PR TITLE
Extend PageMatcher with Environment keyword

### LINKED ISSUE #9612: Add environment (e.g. production) as a new filter to _cascade.target
Hey there,

After looking this up and [confirming](https://discourse.gohugo.io/t/skipping-page-building-depending-on-environment-config/37502) this was not possible (thanks again @jmooring!) I would like to propose a feature, that works similar to the `draft` keyword, but allows specifying the environment(s) for which pages/posts shall be built for. Specifically, I suggest adding an `environment` keyword, which would accept a list of values:

```yaml
environment:
  - production
  - staging
  - development
```

This would allow for per-page/-post management of individual pages/posts, in which individual teams work on parts of a large Hugo deployment, in which none of them (but one admin-team) have access to the actual environment configuration files. Through a keyword inside the pages/posts, teams could individually decide to which environments pages/posts would get deployed to, without having the admin-team involved for adjusting the `cascade._target` configuration.

I would appreciate thoughts on this! Thank you.

### PR DESCRIPTION
**Pull request by CathrinePaulsen**
Fixes #9612 

This fix allows `environment` to be used as a keyword for `cascade._target`. 
It can be used to apply cascade options to the target in only certain build environments, for example to specify in which environment(s) certain pages should not be rendered.

The documentation has been updated with the new keyword.

### CODE REVIEW

File: hugolib/page__meta.go [hunk 1]
```diff
@@ -167,6 +167,10 @@ func (p *pageMeta) Description() string {
 	return p.description
 }
 
+func (p *pageMeta) Environment() string {
```
Comment (bep) [inline hugolib/page__meta.go:170]: 0K, this PR looks mostly good, but we don't want to add the `Environment()` everywhere -- the Page interface is big as it is.

So I suggest that you in the page matcher do something ala:

```go
if m.Environment != "" {
		g, err := glob.GetGlob(m.Environment)
		if err == nil && !g.Match(p.Site().Hugo().Environment()) {
			return false
		}
	}

```

**Review by bep — CHANGES_REQUESTED** [review]

File: hugolib/page__meta.go (hunk 1 above)
Comment (bep) [reply hugolib/page__meta.go:170]: Yea, I see your point. Ultimately we may want to split it (e.g. having both a MatchesSite and a Matches Page method), but even then you will have the same test challenge. 

There is a `testSite` implementation in there somewhere, so I suggest you just expand on that. It makes for a little more work, but certainly a better end game.

Comment (CathrinePaulsen) [reply hugolib/page__meta.go]: I think the most recent change should address the requested change. Let me know what you think!

**Comment by bep** [comment]
Yea, now it looks good, thanks!

**Pull request merged**